code/game/objects/items/weaponry.dm

```diff
@@ -253,18 +253,22 @@ for further reading, please see: https://github.com/tgstation/tgstation/pull/301
 	return(BRUTELOSS)
 
 /obj/item/katana/cursed
-	slot_flags = null
-	item_flags = DROPDEL
+	name = "cursed katana"
+	desc = "Power comes at a price."
+	slot_flags = null  // cannot escape NODROP by putting on belt.
+	var/datum/brain_trauma/magic/stalker/curse
```

Comment:
If my understand is correct, this is going to hard delete if the trauma is removed by something other than dropping the cursed katana (in this case, admin heals, as well as the staff of healing).

You should instead either hold this as a weakref or make the cursed katana stateless, and just run `cure_trauma_type` in `dropped`.